src/models/explorers.ts

```diff
@@ -25,6 +25,7 @@ export type TExplorerParsingFunction = ((data: IParsingFunctionAPI) => Transacti
 
 export interface ExplorerAPI {
   serviceURL?: string | ExplorerURLs;
+  getTransactionServiceURL?: (chain: SupportedChains) => string;
```

Comment:
it makes sense but I'm not sure I like augmenting the API surface on a concept that is represented by 2 different options.

Reply:
Thanks, Your comment really understandable.
I feel this looks complicated to maintain compatibility between the current string/map `serviceURL` and the new function option to specify dynamically by a chain.

Do you have any idea for being simple?

Reply:
I was thinking, how about passing a self executing function that respects the `serviceURL` api. Something like:

```
serviceUrl: getTransactionServiceURL (chain: SupportedChains): string | ExplorerURLs { ... }(chainAsWeKnowIt) // typescript for convenience
```
With using a closure we have access to the data and the function to decide which value to keep. But I am not sure if that works easily in terms of runtime and when the data is available.

Otherwise, if that does not work then I suppose we can make `serviceUrl` to accept a function, which means we need to implement the logic for it in this library (calling it with the data as available). 

Still I believe the consumer should be able to abstract it no? And especially in this case where we are the consumer?

Reply:
That sounds great.
I look that your suggestion make it more simple.
Thanks, I will try it. Please wait.